lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/FunctionalChaosResult.cxx

```diff
@@ -307,6 +313,110 @@ void FunctionalChaosResult::setInvolvesModelSelection(const Bool involvesModelSe
   involvesModelSelection_ = involvesModelSelection;
 }
 
+/* Conditional expectation accessor */
+FunctionalChaosResult FunctionalChaosResult::getConditionalExpectation(const Indices & conditioningIndices) const
+{
+  // Get marginal input sample and distribution
+  const UnsignedInteger inputDimension = inputSample_.getDimension();
+  conditioningIndices.check(inputDimension);
+  const Sample inputSampleMarginal(inputSample_.getMarginal(conditioningIndices));
+  const Distribution inputDistributionMarginal(distribution_.getMarginal(conditioningIndices));
+  
+  // Check independence
+  if (!distribution_.hasIndependentCopula())
+    throw InvalidArgumentException(HERE) << "This class can only manage an independent copula.";
+
+  // Create the conditioned orthogonal basis
+  const String basicClassName(orthogonalBasis_.getImplementation()->getClassName());
+  OrthogonalBasis orthogonalBasisMarginal;
+  if (basicClassName == "OrthogonalProductPolynomialFactory")
+  {
+    const OrthogonalProductPolynomialFactory* p_basis = dynamic_cast<const OrthogonalProductPolynomialFactory*>(orthogonalBasis_.getImplementation().get());
+    orthogonalBasisMarginal = p_basis->getMarginal(conditioningIndices);
+  }
+  else if (basicClassName != "OrthogonalProductFunctionFactory")
+  {
+    const OrthogonalProductFunctionFactory* p_basis = dynamic_cast<const OrthogonalProductFunctionFactory*>(orthogonalBasis_.getImplementation().get());
+    orthogonalBasisMarginal = p_basis->getMarginal(conditioningIndices);
+  }
+  else
+    throw InvalidArgumentException(HERE) << "This class can only manage an OrthogonalProductPolynomialFactory "
+          << "or an OrthogonalProductFunctionFactory "
+          << "but current basis is" << basicClassName;
```

Comment:
it seems you can use the new services from the other PR right instead of these if / else right ?

Reply:
I do not think so. This is because not all orthogonal functions can benefit from this feature: only tensor-product (orthogonal) functions can. I do not know of any mathematical method to create multivariate orthogonal functions which are not based on tensor products, I admit that. But the principle is still true. This could be checked if we had an attribute stating if a given orthogonal function is based on a tensor-product, such as, e.g. `isTensorProduct()`. Would you want me to implement this new method?

Reply:
no, lets check the class names, but still use getMarginal at least it will spare the dynamic casts

Reply:
@mbaudin47 Have a look at SoizeGhanemFactory for a non-tensorized multivariate orthogonal basis

Reply:
This is fixed in 271a037da58754890130951f83d7f7b33635a80c

Reply:
this is much cleaner, thanks